projects/aca-playwright-shared/src/page-objects/components/dataTable/data-table.component.ts

```diff
@@ -238,17 +238,7 @@ export class DataTableComponent extends BaseComponent {
     }
   }
 
-  async selectItem(name: string): Promise<void> {
-    const isSelected = await this.isRowSelected(name);
-    if (!isSelected) {
-      let row = this.getRowByName(name);
-      await row.hover();
-      await row.locator('.mat-mdc-checkbox').click();
-      await row.locator('.mat-mdc-checkbox-checked').waitFor({ state: 'attached' });
-    }
-  }
-
-  async selectMultiItem(...names: string[]): Promise<void> {
+  async selectItems(...names: string[]): Promise<void> {
```

Comment:
if you're cleaning it up I believe we can move those locators to variables (mat-mdc-checkbox...)
also, is the specific timeout needed? cannot be done better i.e. waiting for some element's state?

shouldn't we also keep this if(!isSelected)?

Reply:
all three things fixed. Hope it will work now